test/kaniko_task_test.go

```diff
@@ -0,0 +1,185 @@
+// +build e2e
+
+/*
+Copyright 2018 Knative Authors LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+
+	// Mysteriously by k8s libs, or they fail to create `KubeClient`s from config. Apparently just importing it is enough. @_@ side effects @_@. https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	kanikoTaskName     = "kanikotask"
+	kanikoTaskRunName  = "kanikotask-run"
+	kanikoResourceName = "go-example-git"
+	kanikoBuildOutput  = "Build successful"
+)
+
+func getGitResource(namespace string) *v1alpha1.PipelineResource {
+	return &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kanikoResourceName,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: v1alpha1.PipelineResourceTypeGit,
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/pivotal-nader-ziada/gohelloworld",
+				},
+			},
+		},
+	}
+}
+
+func getTask(namespace string, t *testing.T) *v1alpha1.Task {
+	dockerRepo := os.Getenv("DOCKER_REPO_OVERRIDE")
+	if dockerRepo == "" {
+		t.Fatalf("DOCKER_REPO_OVERRIDE env variable is required")
+	}
+
+	return &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      kanikoTaskName,
+		},
+		Spec: v1alpha1.TaskSpec{
+			Inputs: &v1alpha1.Inputs{
+				Resources: []v1alpha1.TaskResource{
+					v1alpha1.TaskResource{
+						Name: kanikoResourceName,
+						Type: v1alpha1.PipelineResourceTypeGit,
+					},
+				},
+			},
+			BuildSpec: &buildv1alpha1.BuildSpec{
+				Timeout: &metav1.Duration{Duration: 2 * time.Minute},
+				Steps: []corev1.Container{{
+					Name:  "kaniko",
+					Image: "gcr.io/kaniko-project/executor",
+					Args: []string{"--dockerfile=/workspace/Dockerfile",
+						fmt.Sprintf("--destination=%s/kanikotasktest", dockerRepo),
+						"--no-push"},
```

Comment:
i think we should keep using the template here for now, since once we implement our own templating we'll want to come back and change this either way.

but if you prefer to leave this as is then I don't mind! (sorry for going back and forth on this!)

Reply:
can you also include a comment explaining why we are using `--no-push` here?

Reply:
I don't want to switch back to using template then have to remove it again

Reply:
i think it would be good to put some more context into the comment for folks who haven't been following the issue - maybe you could just reference #144 now that we have an issue for it? e.g. something like `TODO(#144) remove the --no-push flag once we can actually push to a registry from our tests on prow`